sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -69,10 +71,12 @@ abstract class QueryGenerator(private val query: BindableQuery) {
       if (bindArg?.isArrayParameter() == true) {
         needsFreshStatement = true
 
-        val indexCalculator = "index + $offset"
-        result.addStatement("""
-          |val ${type.name}Indexes = createArguments(count = ${type.name}.size, offset = $offset)
-        """.trimMargin())
+        if (argument !in seenArrayArguments) {
```

Comment:
i think this can be `if (!seenArrayArguments.add(argument))`?

Reply:
Yep good call, will update. (And fix the shadowed name issue)